src/app/models/saved_function.rb

```diff
@@ -0,0 +1,4 @@
+class SavedFunction < ApplicationRecord
+  belongs_to :user
+  belongs_to :function
```

Comment:
[R] we'll need a validation to prevent user to save the same function twice, you can have
```
validates :function, uniqueness: { scope: :user }
```
or invert it 
```
validates :user, uniqueness: { scope: function }
```

Reply:
I'll go with the first suggestion because it makes sense more than the inverted one